### PR TITLE
Build python bindings in CI via CMake

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

**Comment by marbre** [comment]
@GMNGeoffrey Is `numpy` installed in the CI runner?

**Comment by ScottTodd** [comment]
I think `numpy` is installed - `import numpy as np` from a python3 REPL at least doesn't throw any errors.

Are there more useful error logs that we could pull out from ctest?

```
152/177 Test #173: bindings_python_pyiree_rt_function_abi_test ....***Failed    0.14 sec
153/177 Test #174: bindings_python_pyiree_rt_hal_test .............***Failed    0.14 sec
```

It would be great if we could debug this without needing to SSH into a VM or blindly try things :/

**Comment by marbre** [comment]
@ScottTodd I added `-V` to ctest. This change can be dropped after we fixed the issue. Please rerun kokoro.

**Comment by GMNGeoffrey** [comment]
Aha! These errors look a lot like what I was seeing in https://github.com/google/iree/issues/818

Now that we've seen it in the wild I think it should be a higher priority to fix

```
153: find: ‘/tmpfs/src/github/iree/build/iree_compiler_Translation_Interpreter_Transforms_test_make_executable_abi.mlir.test_test_tmpdir’: No such file or directory
153: find: ‘/tmpfs/src/github/iree/build/iree_compiler_Translation_Interpreter_Transforms_test_xla_slice.mlir.test_test_tmpdir’: No such file or directory
1
```

**Comment by GMNGeoffrey** [comment]
Also Cmake verbose output is a bit hard to follow with multithreaded because it just weaves all the test logs together :-/ It would be nice to have the results separated by test and then reordered.  I suppose something like `ctest -V | grep -E "\d+: " | sort` would do it for a real hacky solution...

**Comment by GMNGeoffrey** [comment]
Also I think the bazel test is failing because you're based of an old commit that still wants to use an older version of bazel (we updated the CI to bazel 2.1)

**Comment by marbre** [comment]
I think it's fine to run ctest single threaded, just to check the output. Shouldn't take that long. Also rebased upon current master.

**Comment by GMNGeoffrey** [comment]
> I think it's fine to run ctest single threaded, just to check the output. Shouldn't take that long. Also rebased upon current master.

Oh just in general it would be nice if the logs actually contained the errors. I haven't turned on verbose mode because it's such a mess in multithreaded

**Comment by marbre** [comment]
Okay, the problem actually is that absl-py isn't installed: `ImportError: No module named 'absl'`

**Comment by marbre** [comment]
> Oh just in general it would be nice if the logs actually contained the errors. I haven't turned on verbose mode because it's such a mess in multithreaded

Is there a possibility to access ` Testing/Temporary/LastTest.log`? That actually contains everything printed also in verbose mode, but I haven't had a look when running ctest in parallel.

**Comment by GMNGeoffrey** [comment]
> > Oh just in general it would be nice if the logs actually contained the errors. I haven't turned on verbose mode because it's such a mess in multithreaded
> 
> Is there a possibility to access ` Testing/Temporary/LastTest.log`? That actually contains everything printed also in verbose mode, but I haven't had a look when running ctest in parallel.

I'll see. Let's create a separate issue for this. I may not get to it this week though

**Comment by marbre** [comment]
> Okay, the problem actually is that absl-py isn't installed: `ImportError: No module named 'absl'`

@GMNGeoffrey Is absl-py installed in the VM?

**Comment by GMNGeoffrey** [comment]
I don't think it is, no. We can add it

**Comment by GMNGeoffrey** [comment]
> I don't think it is, no. We can add it

@rsuderman is looking into adding it to the cmake VMs

**Comment by marbre** [comment]
> > I don't think it is, no. We can add it
> 
> @rsuderman is looking into adding it to the cmake VMs

That would be great :) Thanks Rob.

I think it might be reasonable to enable building the python bindings as an intermediate step, but to disable testing, until we got the CI working.

**Comment by rsuderman** [comment]
The kokoro build bot should be updated with the correct python support (absl installed, upgraded to python3.7).

You will need to add the python bindings tests that use Vulkan to the kokoro exclusion list.
https://github.com/google/iree/blob/master/build_tools/cmake/test.sh#L29

Specifically, you need to exclude all bindings_rt, though the bindings_rt_hal test should be fine to run.

**Comment by GMNGeoffrey** [comment]
Still seeing errors in the cmake CI. I think we'll need to add `IREE_DEFAULT_BACKENDS="tf,iree_interpreter"` to the test environment like in https://github.com/google/iree/blob/master/.github/workflows/bazel_build.yml

Can add it next to IREE_VULKAN_DISABLE here: http://google3/third_party/iree/build_tools/cmake/test.sh

**Comment by marbre** [comment]
Yes, there is still some work left. Add least the absl-py issue is fixed, thanks to @rsuderman and thus e.g. `bindings_python_pyiree_compiler_compiler_test` passes. However, building against the latest master, all python tests will fail in the CMake build. To fix this changes like in #982, but #905 is still unsolved.

**Comment by GMNGeoffrey** [comment]
FYI we've been experiencing issues with the CMake VMs since the image update. Somehow something is deleting the directory they're using while the jobs are running. In an effort to isolate the cause, I'm going to try rolling back the image update and see if I can reproduce with the old image.

**Comment by marbre** [comment]
@GMNGeoffrey Disabled all tests. At least we should now note if building the bindings is broken by future changes.

**Comment by GMNGeoffrey** [comment]
SGTM. Yeah building is still a step in the right direction :-)
